src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  const obj = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(obj, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const prop of action.keysToRemove) {
+          delete obj[prop];
+        }
+        break;
+      case 'clear':
+        for (const key in obj) {
+          delete obj[key];
+        }
+        break;
+    }
+
+    res.push(Object.assign({}, obj));
```

Comment:
Pushing the state copy to the result array should be done outside of the `switch` block, after the current action is processed, to avoid repeating this action in every case.